Misc/NEWS.d/next/Library/2024-12-03-14-45-16.gh-issue-98188.GX9i2b.rst

```diff
@@ -0,0 +1,2 @@
+Fix an issue in :meth:`email.message.Message.get_payload` where data
+cannot be decoded if there is a trailing whitespace and/or texts following the mechanism.
```

Comment:
The parser recognized both "base64" and "base64 balba" as "base64," but in the `get_payload` method, it used the original value from the header with spaces and junk text. In this patch, instead of reading the original header value, it read the parsed value.

Before making this PR, I was also thinking about "base64 blabla" (it's not a valid value based on the RFC), but the current parser can parse "base64 blabla" correctly with defects under default policy. I haven't tested it under strict policy yet, but I agree it should be rejected under strict policy.